tests/test_inner.py

```diff
@@ -1304,3 +1319,34 @@ def test_add_pulp_registry():
 
     push_conf.add_pulp_registry("registry2.example.com", "http://registry2.example.com", True)
     assert len(push_conf.pulp_registries) == 2
+
+
+def test_layer_sizes():
+    flexmock(DockerfileParser, content='df_content')
+    this_file = inspect.getfile(PreRaises)
+    mock_docker()
+    fake_builder = MockInsideBuilder()
+    flexmock(InsideBuilder).new_instances(fake_builder)
+    watch_exit = Watcher()
+    watch_buildstep = Watcher()
+    workflow = DockerBuildWorkflow(SOURCE, 'test-image',
+                                   exit_plugins=[{'name': 'uses_source',
+                                                  'args': {
+                                                      'watcher': watch_exit,
+                                                  }}],
+                                   buildstep_plugins=[{'name': 'buildstep_watched',
+                                                       'args': {
+                                                           'watcher': watch_buildstep,
+                                                       }}],
+                                   plugin_files=[this_file])
+
+    workflow.build_docker_image()
+
+    expected = [
+        {'diff_id': u'sha256:diff_id1', 'size': 4},
+        {'diff_id': u'sha256:diff_id2', 'size': 3},
+        {'diff_id': u'sha256:diff_id3', 'size': 2},
+        {'diff_id': u'sha256:diff_id4', 'size': 1}
```

Comment:
Wrong expectation. We expect sha256:diff_id1 with 1 (oldest layer), sha256:diff_id2 with 2, etc.

Reply:
expectation is good, but because of that wrong history order it seemed wrong